python/cugraph_service/cugraph_service_server/cugraph_handler.py

```diff
@@ -697,30 +785,36 @@ def get_graph_vertex_property_names(self, graph_id):
         if isinstance(G, (PropertyGraph, MGPropertyGraph)):
             return G.vertex_property_names
 
-        raise CugraphServiceError("Graph does not contain properties")
+        return []
 
     def get_graph_edge_property_names(self, graph_id):
         G = self._get_graph(graph_id)
         if isinstance(G, (PropertyGraph, MGPropertyGraph)):
             return G.edge_property_names
 
-        raise CugraphServiceError("Graph does not contain properties")
+        return []
 
     def get_graph_vertex_types(self, graph_id):
         G = self._get_graph(graph_id)
         if isinstance(G, (PropertyGraph, MGPropertyGraph)):
             return G.vertex_types
-
-        raise CugraphServiceError("Graph does not contain properties")
-        # Note: this is currently invalid for a graph without properties
+        else:
+            return [""]
 
     def get_graph_edge_types(self, graph_id):
         G = self._get_graph(graph_id)
         if isinstance(G, (PropertyGraph, MGPropertyGraph)):
             return G.edge_types
-
-        raise CugraphServiceError("Graph does not contain properties")
-        # FIXME this should be valid for a graph without properties
+        else:
+            if G.edgeTypeCol in G.edgelist.edgelist_df.columns:
+                return (
+                    G.edgelist.edgelist_df[G.edgeTypeCol]
+                    .unique()
+                    .astype("str")
+                    .values_host
+                )
+            else:
+                return [""]
```

Comment:
optional, or maybe a FIXME: since cugraph Graphs now have the concept of an edge type, should this code be moved into a `G.get_graph_edge_types()`, or (to match PG) `G.edge_types` ?

Reply:
This should be a fixme with issue; moving it would require some new tests.

Reply:
#2899